### ISSUE TITLE
mesh2fem.barrier fails with latest fmesher

### ISSUE THREAD

**Issue by seananderson** [eliaskrainski/INLAspacetime#3]
As of the latest fmesher on CRAN (0.1.6), `mesh2fem.barrier()` returns the error:

> Error in mesh2fem.barrier(mesh, triBarrier): fm_manifold(mesh, c("S", "R")) is not TRUE

in cases that used to work. Below is an example taken from the INLAspacetime vignette on barriers.

``` r
library(INLA)
#> This is INLA_24.06.14 built 2024-06-14 14:58:21 UTC.
```

``` r
library(INLAspacetime)
#> Loading required package: fmesher
```

``` r
library(sf)
#> Linking to GEOS 3.11.0, GDAL 3.5.3, PROJ 9.1.0; sf_use_s2() is TRUE
```

``` r
rxy <- c(10, 6)
r <- mean(rxy)
bb <- rbind(
  c(0, 1) * rxy[1],
  c(0, 1) * rxy[2]
)
domain.xy <- cbind(
  c(0, 1, 1, 0, 0) * rxy[1],
  c(0, 0, 1, 1, 0) * rxy[2]
)
barrier1 <- cbind(
  cos(seq(0, 2 * pi, length = 50)) * 0.7 + 0.50 * rxy[1],
  sin(seq(0, 2 * pi, length = 50)) * 1.0 + 0.75 * rxy[2]
)
barrier2 <- cbind(
  c(0.50, 1.50, 1.50, 0.50, 0.50) * rxy[1],
  c(0.33, 0.33, 0.40, 0.40, 0.33) * rxy[2]
)
domain <- st_sfc(st_polygon(list(domain.xy)))
barriers <- st_sfc(st_multipolygon(
  list(
    st_polygon(list(barrier1)),
    st_polygon(list(barrier2))
  )
))
mesh <- inla.mesh.2d(
  loc.domain = domain.xy,
  max.edge = c(0.03, 0.1) * r,
  offset = c(0.1, 0.3) * r,
  cutoff = 0.01 * r
)
triBarrier <- unlist(fmesher::fm_contains(
  x = barriers,
  y = mesh,
  type = "centroid"
))
bfem <- mesh2fem.barrier(mesh, triBarrier)
#> Error in mesh2fem.barrier(mesh, triBarrier): fm_manifold(mesh, c("S", "R")) is not TRUE
```

``` r
packageVersion("INLA")
#> [1] '24.6.14'
```

``` r
packageVersion("INLAspacetime")
#> [1] '0.1.8.2'
```

``` r
packageVersion("fmesher")
#> [1] '0.1.6'
```

``` r
sessionInfo()
#> R version 4.4.0 (2024-04-24)
#> Platform: aarch64-apple-darwin20
#> Running under: macOS Sonoma 14.4
#> 
#> Matrix products: default
#> BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib 
#> LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0
#> 
#> locale:
#> [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
#> 
#> time zone: America/Vancouver
#> tzcode source: internal
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> other attached packages:
#> [1] sf_1.0-16               INLAspacetime_0.1.8.002 fmesher_0.1.6          
#> [4] INLA_24.06.14           sp_2.1-4                Matrix_1.7-0           
#> 
#> loaded via a namespace (and not attached):
#>  [1] vctrs_0.6.5        cli_3.6.2          knitr_1.47         rlang_1.1.4       
#>  [5] xfun_0.45          KernSmooth_2.23-24 DBI_1.2.3          purrr_1.0.2       
#>  [9] styler_1.10.3      glue_1.7.0         e1071_1.7-14       htmltools_0.5.8.1 
#> [13] rmarkdown_2.27     R.cache_0.16.0     grid_4.4.0         classInt_0.4-10   
#> [17] evaluate_0.24.0    fastmap_1.2.0      yaml_2.3.8         lifecycle_1.0.4   
#> [21] compiler_4.4.0     fs_1.6.4           Rcpp_1.0.12        rstudioapi_0.16.0 
#> [25] R.oo_1.26.0        R.utils_2.12.3     lattice_0.22-6     digest_0.6.35     
#> [29] class_7.3-22       reprex_2.1.0       parallel_4.4.0     splines_4.4.0     
#> [33] magrittr_2.0.3     R.methodsS3_1.8.2  proxy_0.4-27       tools_4.4.0       
#> [37] withr_3.0.0        units_0.8-5
```

<sup>Created on 2024-06-18 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Comment by finnlindgren** [comment]
Thanks for reporting. I've confirmed this is a regression in `fmesher`, such that it only checks the first of the given manifold type options given. Bug isse added at https://github.com/inlabru-org/fmesher/issues/16

**Comment by finnlindgren** [comment]
This has now been fixed in `fmesher` version 0.1.6.9001. Until I can submit 0.1.7 to CRAN (which will have to wait about 2 weeks due to CRAN rules), installing 0.1.5 or 0.1.6.9001 (from github or r-universe) should work around the bug and allow INLAspacetime to work.

**Comment by finnlindgren** [comment]
`fmesher` version 0.1.7 that fixes this has now been accepted by CRAN and will appear in the repositories in the coming hours&days.
